pocketimdb/users/views.py

```diff
@@ -27,4 +35,33 @@ def create(self, request, *args, **kwargs):
     @action(detail=False, url_path='me', permission_classes=[IsAuthenticated]) 
     def get_current_user(self, request):          
         response_serializer = UserSerializer(request.user)
-        return Response(response_serializer.data, HTTP_200_OK)
\ No newline at end of file
+        return Response(response_serializer.data, HTTP_200_OK)
+
+class WatchlistViewSet(mixins.ListModelMixin,
+                    mixins.DestroyModelMixin,
+                    mixins.UpdateModelMixin,
+                    viewsets.GenericViewSet):
+
+    permission_classes = [IsAuthenticated, UserAccessPermission]
+    pagination_class = None
+
+    def get_serializer_class(self):
+        if self.action == 'partial_update':
+            return UpdateMovieWatchlistSerializer
+        return MovieWatchlistSerializer
+
+    def get_queryset(self):
+        return MovieWatchlist.objects.filter(user=self.kwargs['user_pk']).order_by('id')
+
+    def create(self, request, user_pk):
+        serializer = AddAndRemoveMovieWatchlistSerializer(data=request.data)
+        serializer.is_valid(raise_exception=True)
+        
+        movie_id = serializer.data['movie']
+        if self.get_queryset().filter(movie_id=movie_id).exists():
+            error = {"unique_error": ["Chosen movie is already in your watchlist!"]}
+            return Response(error, status=HTTP_403_FORBIDDEN)
```

Comment:
ovo mi je prije 400, forbidden ces bacati u slucaju da neko ne moze da pristupi nekom resursu, tipa neki korisnik pokusava da pristupi necijoj drugoj wathclist-i, to je za 403, ovo je vise kao pokusao si da dodas film u svoju watchlist-u a vec postoji, mada ovaj slucaj mozemo iskociti, odnosno create_or_update korisimo i onda ce samo update-ovati vec postojeci record, a nece ga kreirati ako je vec kreiran

Reply:
Sustinski sam prvobitno hteo da odradim create_or_update ali kao ajde da ga obavestim da vec postoji taj movie. Ali da, slazem se da nije toliko potrebno bacati gresku vec je okej da bude create_or_update